deploy/crds/infinispan.org_caches_crd.yaml

```diff
@@ -30,9 +30,82 @@ spec:
           type: object
         spec:
           description: CacheSpec defines the desired state of Cache
+          properties:
+            adminAuth:
+              description: Authentication info
+              properties:
+                password:
+                  description: Secret and key containing the admin password for authentication.
+                  properties:
+                    key:
+                      description: The key of the secret to select from.  Must be
+                        a valid secret key.
+                      type: string
+                    name:
+                      description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                        TODO: Add other useful fields. apiVersion, kind, uid?'
+                      type: string
+                    optional:
+                      description: Specify whether the Secret or its key must be defined
+                      type: boolean
+                  required:
+                  - key
+                  type: object
+                username:
+                  description: Secret and key containing the admin username for authentication.
+                  properties:
+                    key:
+                      description: The key of the secret to select from.  Must be
+                        a valid secret key.
+                      type: string
+                    name:
+                      description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                        TODO: Add other useful fields. apiVersion, kind, uid?'
+                      type: string
+                    optional:
+                      description: Specify whether the Secret or its key must be defined
+                      type: boolean
+                  required:
+                  - key
+                  type: object
+              type: object
+            clusterName:
+              description: Name of the cluster where to create the cache
+              type: string
+            name:
+              description: Name of the cache to be created. If empty ObjectMeta.Name
+                will be used
+              type: string
+            template:
+              description: Cache template in XML format
```

Comment:
embedding xml in yaml? 

how about pointing to the location of an xml file instead?

```
template:
  path: /path/to/mycache.xml
```

Reply:
this is something that needs implementation on the oc (or k8s client)

Reply:
maybe the work @tristantarrant has done on the CLI could help us here? if my understanding is correct this CRD lets users wrap infinispan.xml in yaml, which just feels sort of hideous for complex configuration. I think that is for the power users who would be more likely to adopt a CLI approach for the heavy configuration.

to create caches via the Operator through the Infinispan CR maybe it should just be with one of the Infinispan cache templates for the DG service or a copy of the default cache for the Cache service?